Provision Resources to Deploy EPiServerCMS in Azure

rajbalakrish opened the pull request:
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

azurecla commented:
Hi __@rajbalakrish__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

rajbalakrish commented:
Pl. review PR and feedback if everything is OK

azurecla commented:
__@rajbalakrish__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

rajbalakrish commented:
@gatneil Could you review this Pull Request please

gatneil commented:
Please rename the folder to 'episerver'. A full solution like this shouldn't be numbered, and it should be clear from context that it's on Azure :)

gatneil commented:
There are a lot of parameters with no default values; please either give them default values or make them variables. In the case of storage account names, please see the best practice around auto-generating storage account names: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

gatneil commented on 101-episerver-in-azure/azuredeploy.json:
```diff
@@ -0,0 +1,349 @@
+         "type":"Microsoft.Web/serverfarms",
+         "location":"[resourceGroup().location]",
+         "apiVersion":"2014-06-01",
+         "dependsOn":[
```
If there are no dependsOn, can we remove this empty list?

gatneil commented on 101-episerver-in-azure/azuredeploy.json:
```diff
@@ -0,0 +1,349 @@
+         "type":"Microsoft.Sql/servers",
+         "location":"[resourceGroup().location]",
+         "apiVersion":"2014-04-01-preview",
+         "dependsOn":[
```
same

gatneil commented on 101-episerver-in-azure/azuredeploy.json:
```diff
@@ -0,0 +1,349 @@
+         "name":"[parameters('sqlserverName')]",
+         "type":"Microsoft.Sql/servers",
+         "location":"[resourceGroup().location]",
+         "apiVersion":"2014-04-01-preview",
```
is there a non-preview api version we can use?

gatneil commented on 101-episerver-in-azure/azuredeploy.json:
```diff
@@ -0,0 +1,349 @@
+               "name":"AllowAllWindowsAzureIps",
+               "type":"firewallrules",
+               "location":"[resourceGroup().location]",
+               "apiVersion":"2014-04-01-preview",
```
same

gatneil commented on 101-episerver-in-azure/azuredeploy.json:
```diff
@@ -0,0 +1,349 @@
+               "name":"[parameters('SQL-DatabaseName')]",
+               "type":"databases",
+               "location":"[resourceGroup().location]",
+               "apiVersion":"2014-04-01-preview",
```
same

rajbalakrish replied:
Agreed. Will address in the next commit

gatneil commented on 101-episerver-in-azure/azuredeploy.json:
```diff
@@ -0,0 +1,349 @@
+         "type":"string",
+         "value":"[listkeys(variables('defaultAuthRuleResourceId'), variables('serviceBusApiVersion')).primaryKey]"
+      },
+      "DefaultSqlConnectionString":{
```
Are we sure we want this as an output? Not sure if that's good security practice.

rajbalakrish replied:
That's right. Included for local testing and can be removed

gatneil commented on 101-episerver-in-azure/README.md:
```diff
@@ -0,0 +1,20 @@
+
+[Service Bus Pricing](https://azure.microsoft.com/en-us/pricing/details/service-bus/)
+
+For more information about Running EPiServer in Azure, [Click Here](https://azure.microsoft.com/en-us/blog/announcing-episerver-cms-in-azure-marketplace-3/).
```
It looks like this template only deploys the resources required to provision EpiServer, and it doesn't actually deploy episerver itself. Can we add some instructions or a link here on how to deploy episerver once this template has finished?

rajbalakrish replied:
Indeed. I will add brief instructions on Deploying EpiserverCMS

gatneil commented:
Please address the above comments :)

rajbalakrish commented:
@gatneil Thanks for your exhaustive review and hereby fixed the below items. Pl chcek if this is now Ok for a merge :) Thanks for all your patience 

1. Renamed folder name to episerver-cms-in-azure
2. Added Default Value to parameters to minimize user effort
3. Removed the Preview Apiversion and moved to stable Api versions
4. Removed Empty dependsOn List
5. Updated ReadMe with Links leading to step by step guidance on Deploying EpiserverCMS from VisualStudio

gatneil commented on episerver-cms-in-azure/azuredeploy.parameters.json:
```diff
@@ -0,0 +1,36 @@
+      "value": "GEN-UNIQUE"
+    },
+    "sqlserverAdminLogin": {
+      "value": "GEN-USER"
```
GEN-USER isn't an actual placeholder. For users, we usually do something generic like 'azureuser' or use GEN-UNIQUE

rajbalakrish replied:
Agreed :+1: Commit on the way !!!

gatneil commented:
One small change in the parameters file. Otherwise looks good :)

rajbalakrish commented:
@gatneil : Hereby Fixed :100:

gatneil commented:
Looks good; merging; thanks for the contribution! :)

rajbalakrish commented:
Many Thanks for your time and review @gatneil
